src/include/storage/storage_structure/disk_overflow_file.h

```diff
@@ -37,19 +37,25 @@ class DiskOverflowFile : public StorageStructure {
         return copy;
     }
 
+    struct OverflowCache {
+        FileHandle* fileHandle = nullptr;
+        common::page_idx_t pageIdx = UINT32_MAX;
+        uint8_t* frame = nullptr;
+    };
+
     void readStringsToVector(
         transaction::TransactionType trxType, common::ValueVector& valueVector);
     void readStringToVector(transaction::TransactionType trxType, common::ku_string_t& kuStr,
-        common::InMemOverflowBuffer& inMemOverflowBuffer);
+        common::InMemOverflowBuffer& inMemOverflowBuffer, OverflowCache& overflowCache);
 
     inline void scanSingleStringOverflow(
         transaction::TransactionType trxType, common::ValueVector& vector, uint64_t vectorPos) {
+        OverflowCache overflowCache;
```

Comment:
for `scanSingleStringOverflow` I think the general assumption is it's pure random so we should not try to apply cache.